pruntime/enclave/src/lib.rs

```
@@ -1092,6 +1106,41 @@ fn sync_block(input: SyncBlockReq) -> Result<Value, Value> {
     }))
 }
 
+fn sync_events(input: SyncEventsReq) -> Result<Value, Value> {
+	let mut state = STATE.lock().unwrap();
+
+	let mut root: [u8; 32] = Default::default();
+	root.copy_from_slice(&input.root);
+	let state_root = Hash::from(root);
+	let validation = state.light_client.validate_events_proof(&state_root, input.proof, input.events.clone(), input.key);
+	if validation.is_ok() {
+		let events = Vec::<EventRecord<chain::Event, Hash>>::decode(&mut &input.events[..]);
+		match events {
+			Ok(evts) => {
+				for evt in evts {
+					match &evt.event {
+						chain::Event::pallet_phala(be) => {
+							println!("pallet_phala event: {:?}", be);
+							match &be {
+								RawEvent::CommandPushed(w, _c, _p, _n) => {
+									println!("CommandPushed from:{:?}", w);
+								},
+								_ => (),
+							}
+						},
+						_ => (),
+					}
+				}
+			},
+			Err(_) => println!("decode events error"),
+		}
+	}
```

Should return an error if the validation is not ok